gh-actions-scripts/prepare_changed_artifacts.sh

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+
+######################################################################################
+#
+# This script heavily uses Bash parameter expansion.
+# To better understand what is going on here, you can read up on the topic here:
+# https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html
+#
+######################################################################################
+
+CHANGED_FILES=$1
+
+if [ $# -ne 1 ]; then
+  echo "Usage: $0 CHANGED_FILES"
+  exit
+fi
+
+# initialize variables with false (make sure they are also set in needs.prepare_ci_run.outputs !!!)
+BUILD_FUNCTIONS_RUNTIME_SVC=false
+BUILD_LFC_SCHEDULER_SVC=false
+BUILD_OPERATOR_SVC=false
+
+# Set the list of artifacts corresponding to the build that's currently running
+if [ "$BUILD_EVERYTHING" != 'true' ]; then
+  # This is a normal build that needs to build and test the changed artifacts
+  echo "Preparing for build of changed artifacts..."
+  artifacts=(
+    "$FUNCTIONS_RUNTIME_SVC_ARTIFACT_PREFIX"
+    "$LFC_SCHEDULER_SVC_ARTIFACT_PREFIX"
+    "$OPERATOR_SVC_ARTIFACT_PREFIX"
+  )
+elif [ "$BUILD_EVERYTHING" == 'true' ] && [ "$GITHUB_REF_PROTECTED" == 'true' ]; then
+  # This is a build-everything build from a protected branch, i.e. master or a maintenance branch, we should run bridge tests in this case
+  echo "Preparing for master/maintenance branch build..."
+  artifacts=(
+    "$FUNCTIONS_RUNTIME_SVC_ARTIFACT_PREFIX"
+    "$LFC_SCHEDULER_SVC_ARTIFACT_PREFIX"
+    "$OPERATOR_SVC_ARTIFACT_PREFIX"
+  )
+elif [ "$BUILD_EVERYTHING" == 'true' ] && [ "$GITHUB_REF_PROTECTED" != 'true' ]; then
+  # This is a build-everything build for integration tests, which skips bridge code style and UI tests
+  echo "Preparing for build-everything build..."
+  artifacts=(
+    "$FUNCTIONS_RUNTIME_SVC_ARTIFACT_PREFIX"
+    "$LFC_SCHEDULER_SVC_ARTIFACT_PREFIX"
+    "$OPERATOR_SVC_ARTIFACT_PREFIX"
+  )
+fi
```

Comment:
I don't think that we need all those options here. Maybe this could be simplified.